spec/lib/miq_automation_engine/miq_ae_method_spec.rb

```diff
@@ -1,131 +1,143 @@
-      expect(ws.root("encrypted")).to eq('********')
-    end
+    context "needing datatore" do
+      before(:each) do
+        MiqAeDatastore.reset
+        @user = FactoryGirl.create(:user_with_group)
+        @domain = 'SPEC_DOMAIN'
+        @model_data_dir = File.join(File.dirname(__FILE__), "data")
+      end
 
-    it "properly processes instance methods via no inheritance" do
-      EvmSpecHelper.import_yaml_model(File.join(@model_data_dir, "method2"), @domain)
-      MiqAeDatastore.reset_default_namespace
-      setup_methods
-      ws = MiqAeEngine.instantiate("/EVM/SYSTEM/TEST/FOO/test_default", @user)
-      expect(ws).not_to be_nil
-      ws = MiqAeEngine.instantiate("/EVM/SYSTEM/TEST/FOO/test_inline", @user)
-      expect(ws).not_to be_nil
-      # puts ws.to_expanded_xml()
-    end
+      def setup_methods
+        object_class = MiqAeClass.find_by_fqname("#{MiqAeDatastore::DEFAULT_OBJECT_NAMESPACE}/Object")
+        method_data = '$evm.log("info", "Default Class Method")
+          exit MIQ_OK'
+        method_options = {:language => 'ruby', :class_id => object_class.id,
+                          :data => method_data, :location => 'inline'}
+        params = [{:name => 'num1', :priority => 1, :datatype => 'integer'},
+                  {:name => 'num2', :priority => 2, :datatype => 'integer', :default_value => 0}]
+        foo_c = FactoryGirl.create(:miq_ae_method, method_options.merge(:name => 'foo', :scope => 'class'))
+        foo_c.inputs.build(params)
+        foo_c.save!
+        foo_i = FactoryGirl.create(:miq_ae_method, method_options.merge(:name => 'foo', :scope => 'instance'))
+        foo_i.inputs.build(params)
+        foo_i.save!
+        method_options = {:language => 'ruby', :name => 'non_existent',
+                          :scope => 'instance', :location => 'builtin',
+                          :class_id => object_class.id}
+        FactoryGirl.create(:miq_ae_method, method_options)
+      end
 
-    it "properly processes instance methods via relationship via no inheritance" do
-      EvmSpecHelper.import_yaml_model(File.join(@model_data_dir, "method2"), @domain)
-      MiqAeDatastore.reset_default_namespace
-      setup_methods
-      expect { MiqAeEngine.instantiate("/EVM/SYSTEM/TEST/FOO/test_missing_parameter", @user) }.to \
-        raise_error(MiqAeException::MethodParmMissing)
-      expect { MiqAeEngine.instantiate("/EVM/SYSTEM/TEST/FOO/test_non_existent_internal", @user) }.to \
-        raise_error(MiqAeException::MethodNotFound)
-      expect { MiqAeEngine.instantiate("/EVM/SYSTEM/TEST/FOO/test_missing", @user) }.to \
-        raise_error(MiqAeException::MethodNotFound)
-      expect(MiqAeEngine.instantiate("/EVM/SYSTEM/TEST/FOO/test_internal", @user)).not_to be_nil
-      expect(MiqAeEngine.instantiate("/EVM/SYSTEM/TEST/FOO/test_inline", @user)).not_to be_nil
-    end
+      it "integrates with base methods of MiqAeServiceModelBase" do
+        vm_name = 'fred flintstone'
```

Comment:
I love the values @tinaafitz picks for variables, at least I think it's her.

Fred, Wilma, Barney... 👍